test/test_type_info.py

```diff
@@ -88,6 +91,26 @@ def test_finfo(self):
         self.assertEqual(xinfo.resolution, 1.0)
         self.assertEqual(xinfo.dtype, "float8_e4m3fn")
 
+    def test_to_complex(self):
+        # Regression test for https://github.com/pytorch/pytorch/issues/124868
+        # If reference count is leaked this would be a set of 10 elements
+        ref_cnt = {sys.getrefcount(torch.float32.to_complex()) for _ in range(10)}
+        self.assertLess(len(ref_cnt), 3)
```

Comment:
Why isn't this equal to 1 ?

Reply:
Because we can run multiple tests in parallel  that theoretically can affect the refcount for the type
But if assumed that testsuite is executed sequentially then yes, it should be equal to one

Reply:
We run the test suite in parallel in multiple threads?? I don't expect that would work well given how heavily we use global states

Reply:
We've been weeding out global state from test for quite a while, and we have a list of tests that should not run in parallel, see https://github.com/pytorch/pytorch/blob/3d1dd79b8092d0bdb060aedd9e1eaeaa2a6e32d8/test/run_test.py#L193-L200